### PR TITLE
AT-80

### PR DESCRIPTION
**Pull request by OllieVanD**
Refactored, finished off Jasmine Feature tests and added Unit test for Header Checker

Co-Author-By: BennyC

### CODE REVIEW

**Review by benclayton98 — APPROVED** [review]
Fantastic work, Oliver

File: spec/unit/file-uploader/headerChecker.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  it("checking that the array checker can returns false if the arrays are not identical", () => {
+    const array2 = ["differentHeaders", "notMatching"]; 
+    const result = checkArraysAreEqual(headerMocks, array2);
+    expect(result).toEqual(false);
+  });
+  it("checking that the array checker can return false if the second array isn't an array.", () => {
+    const fakeObj = {fake: headerMocks};
+    const result = checkArraysAreEqual(headerMocks, fakeObj);
+    expect(result).toEqual(false);
+  });
```
Comment (dearshrewdwit) [inline spec/unit/file-uploader/headerChecker.spec.js:36]: Nice - but move these to a separate spec file for the utility functions

File: src/fileUpload/utility.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 function checkArraysAreEqual (arr1, arr2) {
-  if (typeof arr2 !== "object") {
+  if (!Array.isArray(arr2)) {
```
Comment (dearshrewdwit) [inline src/fileUpload/utility.js:2]: check both?

```js
[arr1, arr2].every(array => Array.isArray(array))
```

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
Nice. Small cleanups

**Review by benclayton98 — APPROVED** [review]
He's only gone and done it again

File: spec/unit/file-uploader/utlitity.spec.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const {checkArraysAreEqual} = require("../../../src/fileUpload/utility.js");
```
Comment (dearshrewdwit) [inline spec/unit/file-uploader/utlitity.spec.js:1]: Wait, what's the name of this spec again? Ut... Util.. Utlit 😂

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

**Comment by dearshrewdwit** [comment]
@benclayton98 A clearly thorough reviewer 😂

File: spec/unit/file-uploader/utlitity.spec.js (hunk 3 above)
Comment (OllieVanD) [reply spec/unit/file-uploader/utlitity.spec.js:1]: Isn't that how you spell it?

Comment (dearshrewdwit) [reply spec/unit/file-uploader/utlitity.spec.js:1]: `spec/unit/file-uploader/utlitity.spec.js` 👀

**Review by dearshrewdwit — APPROVED** [review]

**Pull request merged**